backend/CircleForms/Services/PostsService.cs

```diff
@@ -352,4 +396,67 @@ private async Task<PostRedis[]> MapWithAnswerCounts(PostRedis[] posts)
 
         return posts;
     }
+
+    public async Task<Result<Post>> Publish(string id, string claim)
+    {
+        var postResult = await GetAndValidate(id, claim);
+        if (postResult.IsError)
+        {
+            return postResult;
+        }
+
+        var post = postResult.Value;
+        if (post.Published)
+        {
+            return post;
+        }
+
+        post.Published = true;
+        post.PublishTime = DateTime.UtcNow;
+        await _postRepository.Update(post.ID, post);
+        if (post.Accessibility == Accessibility.Public)
```

Comment:
I believe we agreed on caching all of the posts instead?

Reply:
nope